packages/synapse-react-client/src/components/entity/metadata/AnnotationsTable.tsx

```diff
@@ -82,15 +114,32 @@ export function AnnotationsTable(props: AnnotationsTableProps) {
       <table className="AnnotationsTable">
         <tbody>
           {annotations &&
-            Object.keys(annotations).map((key: string) => {
+            Object.keys(annotations).map((annotationKey: string) => {
               return (
-                <tr key={key} className="AnnotationsTable__Row">
-                  <td className="AnnotationsTable__Row__Key">{key}</td>
+                <tr key={annotationKey} className="AnnotationsTable__Row">
+                  <td className="AnnotationsTable__Row__Key">
+                    <EntityAnnotationKey
+                      annotationKey={annotationKey}
+                      title={getDisplayedAnnotationTitle(
+                        annotationKey,
+                        annotatedInstance,
+                      )}
+                      description={getDisplayedAnnotationDescription(
+                        annotationKey,
+                        annotatedInstance,
+                      )}
+                    />
```

Comment:
New component handles rendering an annotation key that may have a title and/or description